test/lib/ufe/testRename.py

```diff
@@ -380,3 +398,48 @@ def testRenameSpecialCharacter(self):
         cmds.rename(newNameStartingWithDigit)
 
         self.assertFalse(usdPrim.GetName()[0].isdigit())
+
+    def testRenameNotifications(self):
+        '''Rename a USD node and test for the UFE notifications.'''
+
+        # open usdCylinder.ma scene in test-samples
+        mayaUtils.openCylinderScene()
+
+        # clear selection to start off
+        cmds.select(clear=True)
+
+        # select a USD object.
+        mayaPathSegment = mayaUtils.createUfePathSegment('|world|mayaUsdTransform|shape')
+        usdPathSegment = usdUtils.createUfePathSegment('/pCylinder1')
+        cylinderPath = ufe.Path([mayaPathSegment, usdPathSegment])
+        cylinderItem = ufe.Hierarchy.createItem(cylinderPath)
+
+        ufe.GlobalSelection.get().append(cylinderItem)
+
+        # get the USD stage
+        stage = mayaUsd.ufe.getStage(str(mayaPathSegment))
+
+        # set the edit target to the root layer
+        stage.SetEditTarget(stage.GetRootLayer())
+        self.assertEqual(stage.GetEditTarget().GetLayer(), stage.GetRootLayer())
+
+        # Create our UFE notification observer
+        ufeObs = TestObserver()
+
+        # We start off with no observers
+        self.assertFalse(ufe.Scene.hasObjectAddObserver(ufeObs))
+        self.assertFalse(ufe.Scene.hasObjectDeleteObserver(ufeObs))
+        self.assertFalse(ufe.Scene.hasObjectPathChangeObserver(ufeObs))
+
+        # Add the UFE observers we want to test
+        ufe.Scene.addObjectAddObserver(ufeObs)
+        ufe.Scene.addObjectDeleteObserver(ufeObs)
+        ufe.Scene.addObjectPathChangeObserver(ufeObs)
+
+        # rename
+        newName = 'pCylinder1_Renamed'
+        cmds.rename(newName)
+
+        # After the rename we should have 1 rename notif and no unexepected notifs.
+        self.assertEqual(ufeObs.notifications(), 1)
+        self.assertFalse(ufeObs.receivedUnexpectedNotif())
```

Comment:
Here we rename a UFE item. We should get a single UFE rename notif and no unexpected notifs (object add or delete). The test passes now (with my fix) and I ran it without my fix (above) and it failed (as it should).